angular-rails-app/app/controllers/owners_controller.rb

```diff
@@ -3,10 +3,15 @@ class OwnersController < ApplicationController
    before_action :set_owner, only: [:show]
 
   def index
-    @owners = Owner.all
+    if params[:name]
+       @owners = Owner.where("name LIKE (?)", "%#{params[:name]}%").limit(6)
```

Comment:
PS - I think my preference would be to merge this as is for the moment, and create a separate issue of addressing the corporation name question.  But let me know your thoughts.